features/collection/api/src/main/java/org/opennms/netmgt/collection/support/ObjectNameStorageStrategy.java

```diff
@@ -47,7 +47,7 @@ public ObjectNameStorageStrategy() {
     @Override
     public void updateContext(JexlContext context, CollectionResource resource) {
         try {
-            ObjectName oname = new ObjectName(resource.getInstance());
+            ObjectName oname = new ObjectName(resource.getUnmodifiedInstance());
```

Comment:
Can we add some test coverage for this?

Reply:
I had added opennms-dao/src/test/java/org/opennms/netmgt/dao/support/ObjectNameStorageStrategyTest.java back when ObjectNameStorageStrategy was added. Are there test cases that should be inside it?

Reply:
I was thinking of something inline that would cover this particular problem. i.e. the test case should fail if this line was reverted

Reply:
I see why the unit test didn't fail earlier.

1. ObjectNameStorageStrategyTest uses MockCollectionResource which wasn't modifying the instance string.

My last commit, 8c0ca79593274b109624af4d9c8fcf5576ab5211, I also updated it to sanitize the instance string. With this, if I change MockCollectionResource:getUnmodifiedInstance() to return the modified string, the unit tests start to throw org.apache.commons.jexl2.UnifiedJEXL$Exception parse exceptions.

2. In ObjectNameStorageStrategy method updateContext(), a javax.management.MalformedObjectNameException is caught, but the exception is only logged at DEBUG. Should i change this to ERROR?

updateContext() doesn't currently have a throws exception, so I'd also have to modify the method signature in JexlIndexStorageStrategy too.

Reply:
I've modified updateContext() to throw an IllegalArgumentException if we get a MalformedObjectNameException. There is no generic StorageStrategyException, or I would have used that.

I've also added a unit test case to explicitly test for if we try passing the sanitized instance string as the unmodified instance string.

I'd like to rebase & squash all commits before fully merging.